test/unit/gotcha_unit_tests.c

```diff
@@ -221,11 +237,85 @@ Suite* gotcha_libc_suite(){
   tcase_add_test(libc_case, gotcha_heap_test);
   tcase_add_test(libc_case, gotcha_memcpy_test);
   tcase_add_test(libc_case, gotcha_strcmp_test);
+  tcase_add_test(libc_case, gotcha_atoi_test);
+  suite_add_tcase(s, libc_case);
+  return s;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////GOTCHA Auxv Tests///////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+START_TEST(vdso_map_test){
+  struct link_map* hopefully_vdso = get_vdso_from_maps();
+  struct link_map* hopefully_also_vdso = get_vdso_from_auxv();
+  // intentionally not testing aliases explicitly as many systems don't implement this
+  ck_assert_msg(hopefully_vdso, "VDSO not found in maps");
+  ck_assert_msg(hopefully_also_vdso, "VDSO not found in auxv");
+}
+END_TEST
+
+START_TEST(vdso_pagesize_test){
+  int vdso_pagesize = get_auxv_pagesize();
+  ck_assert_msg(vdso_pagesize, "VDSO does not contain page size");
+}
+END_TEST
+
+Suite* gotcha_auxv_suite(){
+  Suite* s = suite_create("Gotcha Auxv");
+  TCase* libc_case = tcase_create("Basic tests");
+  tcase_add_test(libc_case, vdso_map_test);
+  tcase_add_test(libc_case, vdso_pagesize_test);
+  suite_add_tcase(s, libc_case);
+  return s;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////GOTCHA Hash Tests///////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#define NUM_INSERTS 16384
+#define TABLE_SIZE (NUM_INSERTS/2)
+
+START_TEST(hash_grow_test){
+   hash_table_t table;
+   create_hashtable(&table,4,(hash_func_t) strhash, (hash_cmp_t) gotcha_strcmp);
+   char* hashable_string = "dogs";
+   char* pointer_list[NUM_INSERTS];
+   hash_hashvalue_t dogs_val = strhash(hashable_string);  
+   int loop;
+   for(loop=0;loop<NUM_INSERTS;loop++){
+     pointer_list[loop] = (char*)gotcha_malloc(sizeof(char)*2);
+     pointer_list[loop][0] = (char)(loop+1);
+     pointer_list[loop][1] = (char)(0);
+   }
+   for(loop=0;loop<NUM_INSERTS;loop++){
+     addto_hashtable(&table,&pointer_list[loop],hashable_string);
```

Comment:
I've updated this and it breaks, don't know what the contract is for lookup, so I'll wait for your feedback